### ISSUE TITLE
Underworld error - detecting yourself 

### ISSUE THREAD

**Issue by FrutyX** [DinoDevs/GladiatusCrazyAddon#368]
Taken from #311 
--

This one is appearing when your underworld copy drops something, GCA fails to detect the enemy and throws an error.

![bug](https://user-images.githubusercontent.com/23278552/175813022-c4e68334-b66e-4af3-b293-8e38f98c7649.PNG)

![bug2](https://user-images.githubusercontent.com/23278552/175813112-783d4325-e12e-4068-b70c-f8d6da656289.PNG)

`<div id="costumeAnimation_62b849e548476" class="avatar avatar_costume_part avatar_costume_animation" style="z-index: 0; background-image: url("9425/img/costumes/sets/male/3_complete.png"); background-position: -1848px 0px;"></div>`

It also seems it's only for secondary drops, primary was fine:

![bug3](https://user-images.githubusercontent.com/23278552/175813263-fc54027d-60b9-4314-8989-52ca1d4d897e.PNG)

No error here.

> GramThanos: Its the addon trying to detect the materials drioped by the enemies. Here the enemy is yourself, thus it doesn't have an NPC number.

**Comment by FrutyX** [comment]
@GreatApo & @GramThanos when you'll be fixing the currently bugged enemy detection script, could you also check this and adjust it to it? Two for one.

Or perhaps it will fix itself by reworking it,  depends on how you rework it.

**Comment by FrutyX** [comment]
A bit problematic to test it, I will see tomorrow when I am back in UW.

**Comment by FrutyX** [comment]
I am afraid that with this change ( [c551889](https://github.com/DinoDevs/GladiatusCrazyAddon/commit/c551889fb00d69d0fb6f419c8f9bef67f5824b0e) ), you created errors in even in regular reports (you two are brothers, aren't you 😄).

I have tons of these errors:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'replace')

https://s34-cz.gladiatus.gameforge.com/game/index.php?mod=reports&submod=showCombatReport&t=0&reportId=8397465&sh=

core/source/reports.js?4.3.6&built=1658387951508:207
```

and it happens in every report, where there is secondary drop:

![sec](https://user-images.githubusercontent.com/23278552/180211042-0ca8b183-9962-4c39-82b8-3f317ed8568f.PNG)

But the fact that detect enemies is still bugged (reverse lookup url errors) isn't really helping too I guess.

**Comment by GreatApo** [comment]
> I am afraid that with this change ( [c551889](https://github.com/DinoDevs/GladiatusCrazyAddon/commit/c551889fb00d69d0fb6f419c8f9bef67f5824b0e) ), you created errors even in regular reports (you two are brothers, aren't you 😄).
> 

Yes, sorry, that's on me. Yes, we are brothers :)

**Comment by FrutyX** [comment]
@GreatApo But I have some good news too, I just dropped it in UW and haven't got any error, also I did not wear any costume.

![udw](https://user-images.githubusercontent.com/23278552/180213417-f8cfd6f0-0232-461b-9acc-53a7004672e4.PNG)

However, there is some symbol stuff in the resource image :D

**Comment by FrutyX** [comment]
If helpful, this is the avatar URL:

`gf2.geo.gfsrv.net/cdn12/124a63795958541c73a4e72afa1eb1.jpg`

**Comment by GramThanos** [comment]
My code is flawless, I caused no errors.

**Comment by FrutyX** [comment]
Of course Thanos, always is.

**Comment by FrutyX** [comment]
Wasted my UW points, but I can confirm that the fix was actually working, with or without an costume only there are two new problems:

- Without a costume, there are strange symbols in the material image (screenshot above)
- Regular reports (non-UW) are throwing an "replace" error when a secondary drop is dropped.

**Comment by FrutyX** [comment]
Patched in v4.3.6, closing, re-open if needed.
